rbd/mirror_group.go

```diff
@@ -0,0 +1,493 @@
+  return ((rbd_mirror_group_resync_fn) fn)(p, name);
+}
+
+// rbd_mirror_group_get_info_fn matches rbd_mirror_group_get_info function.
+typedef int(*rbd_mirror_group_get_info_fn)(rados_ioctx_t p, const char *name,
+                                           rbd_mirror_group_info_t *mirror_group_info,
+                                           size_t info_size);
+
+// rbd_mirror_group_get_info_dlsym take *fn as rbd_mirror_group_get_info_fn and calls the dynamically loaded
+// rbd_mirror_group_get_info function passed as 1st argument.
+static inline int rbd_mirror_group_get_info_dlsym(void *fn, rados_ioctx_t p, const char *name,
+                                                  rbd_mirror_group_info_t *mirror_group_info,
+                                                  size_t info_size){
+  // cast function pointer fn to rbd_mirror_group_get_info and call the function
+  return ((rbd_mirror_group_get_info_fn) fn)(p, name, mirror_group_info, info_size);
+}
+
+// rbd_mirror_group_get_global_status_fn matches rbd_mirror_group_get_global_status function.
+typedef int(*rbd_mirror_group_get_global_status_fn)(rados_ioctx_t p, const char *name,
+                                                    rbd_mirror_group_global_status_t *mirror_group_status,
+                                                    size_t status_size);
+
+// rbd_mirror_group_get_global_status_dlsym take *fn as rbd_mirror_group_get_global_status_fn and calls the dynamically loaded
+// rbd_mirror_group_get_global_status function passed as 1st argument.
+static inline int rbd_mirror_group_get_global_status_dlsym(void *fn, rados_ioctx_t p, const char *name,
+                                                           rbd_mirror_group_global_status_t *mirror_group_status,
+														   size_t status_size){
+  // cast function pointer fn to rbd_mirror_group_get_global_status and call the function
+  return ((rbd_mirror_group_get_global_status_fn) fn)(p, name, mirror_group_status, status_size);
+}
+*/
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+
+	"github.com/ceph/go-ceph/internal/cutil"
+	"github.com/ceph/go-ceph/internal/dlsym"
+	"github.com/ceph/go-ceph/rados"
+)
+
+// MirrorGroupEnable will enable mirroring for a group using the specified mode.
+//
+// Implements:
+//
+//	int rbd_mirror_group_enable(rados_ioctx_t p, const char *name,
+//	  							rbd_mirror_image_mode_t mirror_image_mode,
+//									uint32_t flags);
+func MirrorGroupEnable(groupIoctx *rados.IOContext, groupName string, mode ImageMirrorMode) error {
+	cGroupName := C.CString(groupName)
+	defer C.free(unsafe.Pointer(cGroupName))
+
+	rbdMirrorGroupEnable, rbdMirrorGroupEnableErr := dlsym.LookupSymbol("rbd_mirror_group_enable")
+	if rbdMirrorGroupEnableErr != nil {
+		return fmt.Errorf("%w: %w", ErrNotImplemented, rbdMirrorGroupEnableErr)
+	}
+
+	ret := C.rbd_mirror_group_enable_dlsym(
+		rbdMirrorGroupEnable,
+		cephIoctx(groupIoctx),
+		cGroupName,
+		C.rbd_mirror_image_mode_t(mode),
+		(C.uint32_t)(2),
```

Comment:
define a const for the flag and add documentation on what each value mean

Reply:
Done